tests/bluetooth/common/testlib/include/testlib/scan.h

```diff
@@ -0,0 +1,12 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_TESTS_BLUETOOTH_COMMON_TESTLIB_INCLUDE_TESTLIB_SCAN_H_
+#define ZEPHYR_TESTS_BLUETOOTH_COMMON_TESTLIB_INCLUDE_TESTLIB_SCAN_H_
+
+#include <zephyr/bluetooth/bluetooth.h>
+
+int bt_testlib_scan_find_name(bt_addr_le_t *result, const char *name);
```

Comment:
What about the suggestion to reverse the order parameters? IMO it makes more sense to have `<param, result>` than `<result, param>`

Arguably since this is just moving existing functionality, it is OK as is.

Reply:
Let's discuss this and I'll make a followup PR. My function `bt_testlib_att_read_by_type_sync` also mutates/outputs to the initial arguments. So we'll have to change multiple prototypes to be consistent.

I'm not sure right now what the best order is. Often the mutated arguments come first (like when adding to a list). Maybe we could draft up a rule in our coding guidelines?

Reply:
Another example is the classic function `memcpy` that writes to its first argument.